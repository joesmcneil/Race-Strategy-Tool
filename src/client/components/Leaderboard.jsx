import { useEffect, useState } from "react";
import LeaderboardItem from "./LeaderboardItem";

const Leaderboard = (props) => {
  const [leaderboardArr, setLeaderboardArr] = useState([]);

  useEffect(() => {
    if (props.status === true) {
      console.log(props.racers);
      const tempArray = [];
      tempArray.push(
        <div className="child"> {/** child of outer flexbox in leaderboard component */}
          <div className="parent"> {/** el row */}
            <div className="child"> {/** el data */}
              Position
            </div>
            <div className="child">
              Alias
            </div>
            <div className="child">
              Interval
            </div>
          </div>
        </div>,
      );

      for (const racer of props.racers) {
        console.log(racer);
        console.log(leaderboardArr);
        tempArray.push(
            <LeaderboardItem
              alias = {racer.alias}
              position = {racer.position}
              timeDelta = {racer.interval}
            />,
        );
        console.log(leaderboardArr);
      }
      setLeaderboardArr(tempArray);
    }
  }, [props.status, props.racers]);

  return (
    <div className="leaderboardParent">
        {leaderboardArr}
    </div>
  );
};

export default Leaderboard;
