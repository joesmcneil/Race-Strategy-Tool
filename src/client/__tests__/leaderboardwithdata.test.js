import renderer from 'react-test-renderer';
import Leaderboard from "../components/Leaderboard";

// Testing the leaderboard component by passing it an array of data and a status of true

it('should render when passed an array containing data', () => {
    
    const status = true;
    const racers = [
        {
            position: 1,
            alias: 'HAM',
            timeDelta: 0
          }
        ]

    const expectedComponent = renderer.create(
    <Leaderboard 
      status={status} 
      racers={racers} 
    />).toJSON();

    expect(expectedComponent).toMatchSnapshot();
})