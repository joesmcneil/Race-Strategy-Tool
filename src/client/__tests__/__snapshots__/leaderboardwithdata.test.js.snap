// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly with an empty array 1`] = `
<div
  className="leaderboardParent"
/>
`;

exports[`testing that the leaderboard component renders properly with an empty array 1`] = `
<div
  className="leaderboardParent"
/>
`;
